Fix a few unit tests

job: https://travis-ci.com/qbox/kubernetes/jobs/164822890 ([log](https://api.travis-ci.com/v3/job/164822890/log.txt))

test 1:

```
--- FAIL: TestSyncPodsStartPod (0.00s) 
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1888257]

goroutine 94 [running]:
testing.tRunner.func1(0xc4207e0e10)
    /usr/local/go/src/testing/testing.go:711 +0x2d2
panic(0x1bc9780, 0x2f64680)
    /usr/local/go/src/runtime/panic.go:491 +0x283
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncPod(0xc420627000, 0x0, 0xc420408a80, 0x0, 0xc420270000, 0x0, 0x0, 0x0)
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1675 +0x1447 
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).(k8s.io/kubernetes/pkg/kubelet.syncPod)-fm(0x0, 0xc420408a80, 0x0, 0xc420270000, 0x0, 0x0, 0x7f67a906b000)
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:865 +0x51
k8s.io/kubernetes/pkg/kubelet.(*fakePodWorkers).UpdatePod(0xc42035f0b0, 0xc42039a1e0)
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/pod_workers_test.go:49 +0x1af
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).dispatchWork(0xc420627000, 0xc420408a80, 0x0, 0x0, 0xbefc9340a670fbf1, 0x14532764f, 0x3083600)
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:2008 +0x157
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).HandlePodSyncs(0xc420627000, 0xc42000e660, 0x1, 0x1)
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:2131 +0xe8
k8s.io/kubernetes/pkg/kubelet.TestSyncPodsStartPod(0xc4207e0e10)
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_test.go:373 +0x2eb
testing.tRunner(0xc4207e0e10, 0x1f65b28)
    /usr/local/go/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
    /usr/local/go/src/testing/testing.go:789 +0x2de
FAIL    k8s.io/kubernetes/pkg/kubelet   5.474s
```

test 2:

```
--- FAIL: TestFindAndAddNewPods_FindAndRemoveDeletedPods (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x12b0246]

goroutine 32 [running]:
testing.tRunner.func1(0xc4204260f0)
    /usr/local/go/src/testing/testing.go:711 +0x2d2
panic(0x1436da0, 0x1ff0770)
    /usr/local/go/src/runtime/panic.go:491 +0x283
k8s.io/kubernetes/pkg/kubelet/volumemanager/populator.(*desiredStateOfWorldPopulator).findAndRemoveDeletedPods(0xc420340410)
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/volumemanager/populator/desired_state_of_world_pop    ulator.go:208 +0x576
k8s.io/kubernetes/pkg/kubelet/volumemanager/populator.TestFindAndAddNewPods_FindAndRemoveDeletedPods(0xc4204260f0)
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/volumemanager/populator/desired_state_of_world_pop    ulator_test.go:115 +0x6e8
testing.tRunner(0xc4204260f0, 0x1690590)
    /usr/local/go/src/testing/testing.go:746 +0xd0 
created by testing.(*T).Run
    /usr/local/go/src/testing/testing.go:789 +0x2de
FAIL    k8s.io/kubernetes/pkg/kubelet/volumemanager/populator   0.083s
```

--- a/pkg/kubelet/apis/logplugin/v1alpha1/constants.go
+++ b/pkg/kubelet/apis/logplugin/v1alpha1/constants.go
@@ -18,7 +18,7 @@ package logplugin
 
 const (
 	// Current version of the API supported by kubelet
-	Version = "v1alpha1"
+	Version       = "v1alpha1"
 	LogPluginPath = "/var/lib/kubelet/log-plugins/"
 	// KubeletSocket is the path of the Kubelet registry socket
 	KubeletSocket = LogPluginPath + "kubelet.sock"

--- a/pkg/kubelet/apis/logplugin/v1beta1/BUILD
+++ b/pkg/kubelet/apis/logplugin/v1beta1/BUILD
@@ -11,7 +11,7 @@ go_library(
         "api.pb.go",
         "constants.go",
     ],
-    importpath = "k8s.io/kubernetes/pkg/kubelet/apis/logplugin/v1alpha1",
+    importpath = "k8s.io/kubernetes/pkg/kubelet/apis/logplugin/v1beta1",
     deps = [
         "//vendor/github.com/gogo/protobuf/gogoproto:go_default_library",
         "//vendor/github.com/gogo/protobuf/proto:go_default_library",

--- a/pkg/kubelet/apis/logplugin/v1beta1/constants.go
+++ b/pkg/kubelet/apis/logplugin/v1beta1/constants.go
@@ -18,7 +18,7 @@ package logplugin
 
 const (
 	// Current version of the API supported by kubelet
-	Version = "v1beta1"
+	Version       = "v1beta1"
 	LogPluginPath = "/var/lib/kubelet/log-plugins/"
 	// KubeletSocket is the path of the Kubelet registry socket
 	KubeletSocket = LogPluginPath + "kubelet.sock"

--- a/pkg/kubelet/kubelet_test.go
+++ b/pkg/kubelet/kubelet_test.go
@@ -73,6 +73,8 @@ import (
 	volumetest "k8s.io/kubernetes/pkg/volume/testing"
 	"k8s.io/kubernetes/pkg/volume/util/volumehelper"
 	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+	logmanager "k8s.io/kubernetes/pkg/kubelet/log/manager"
+	logpolicy "k8s.io/kubernetes/pkg/kubelet/log/policy"
 )
 
 func init() {
@@ -293,6 +295,8 @@ func newTestKubeletWithImageList(
 		NewInitializedVolumePluginMgr(kubelet, kubelet.secretManager, kubelet.configMapManager, []volume.VolumePlugin{plug}, prober)
 	require.NoError(t, err, "Failed to initialize VolumePluginMgr")
 
+	logPolicyStatusManager := logpolicy.NewPolicyStatusManager()
+
 	kubelet.mounter = &mount.FakeMounter{}
 	kubelet.volumeManager = kubeletvolume.NewVolumeManager(
 		controllerAttachDetachEnabled,
@@ -305,7 +309,7 @@ func newTestKubeletWithImageList(
 		kubelet.mounter,
 		kubelet.getPodsDir(),
 		kubelet.recorder,
-		kubelet.logPluginManager,
+		logPolicyStatusManager,
 		false, /* experimentalCheckNodeCapabilitiesBeforeMount*/
 		false /* keepTerminatedPodVolumes */)
 
@@ -316,6 +320,18 @@ func newTestKubeletWithImageList(
 	kubelet.AddPodSyncLoopHandler(activeDeadlineHandler)
 	kubelet.AddPodSyncHandler(activeDeadlineHandler)
 	kubelet.gpuManager = gpu.NewGPUManagerStub()
+
+	logPluginManager, err := logmanager.NewLogPluginManagerImpl(
+		fakeKubeClient,
+		kubelet.recorder,
+		kubelet.podManager,
+		kubelet.configMapManager,
+		kubelet.volumeManager,
+		logPolicyStatusManager,
+	)
+	assert.NoError(t, err)
+	kubelet.logPluginManager = logPluginManager
+
 	return &TestKubelet{kubelet, fakeRuntime, mockCadvisor, fakeKubeClient, fakeMirrorClient, fakeClock, nil, plug}
 }
 

--- a/pkg/kubelet/log/api/util/util_test.go
+++ b/pkg/kubelet/log/api/util/util_test.go
@@ -1,7 +1,6 @@
 package util
 
 import (
-	"k8s.io/kubernetes/pkg/kubelet/util/format"
 	"reflect"
 	"testing"
 
@@ -11,6 +10,7 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/log/api"
 	"k8s.io/kubernetes/pkg/kubelet/log/api/v1alpha1"
 	"k8s.io/kubernetes/pkg/kubelet/log/api/v1beta1"
+	"k8s.io/kubernetes/pkg/kubelet/util/format"
 )
 
 var testcases = []struct {
@@ -378,7 +378,8 @@ func TestIsPodLogPolicyExists(t *testing.T) {
 func TestGetPodLogPolicy(t *testing.T) {
 	for _, tc := range testcases {
 		actual, _ := GetPodLogPolicy(tc.pod)
-		if !reflect.DeepEqual(actual, tc.logPolicy) {
+		// because of conversion from map to slice, use custom deep equal here.
+		if !podLogPolicyDeepEqual(actual, tc.logPolicy) {
 			t.Errorf("test GetPodLogPolicy error, expected: %v, actual: %v, pod: %q", tc.logPolicy, actual, format.Pod(tc.pod))
 		}
 	}
@@ -391,4 +392,31 @@ func TestGetPodLogConfigMapNames(t *testing.T) {
 			t.Errorf("test GetPodLogConfigMapNames error, expected: %v, actual: %v, pod: %q", tc.configMapNames, actual, format.Pod(tc.pod))
 		}
 	}
+}
+
+func podLogPolicyDeepEqual(x *api.PodLogPolicy, y *api.PodLogPolicy) bool {
+	if x == nil || y == nil {
+		return x == y
+	}
+	if x.PluginName != y.PluginName {
+		return false
+	}
+	if x.SafeDeletionEnabled != y.SafeDeletionEnabled {
+		return false
+	}
+	if len(x.ContainerLogPolicies) != len(y.ContainerLogPolicies) {
+		return false
+	}
+
+	// slice compare
+	xm := make(map[api.ContainerLogPolicy]int)
+	ym := make(map[api.ContainerLogPolicy]int)
+	for _, clp := range x.ContainerLogPolicies {
+		xm[clp]++
+	}
+	for _, clp := range y.ContainerLogPolicies {
+		ym[clp]++
+	}
+
+	return reflect.DeepEqual(xm, ym)
 }

--- a/pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go
+++ b/pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go
@@ -37,6 +37,7 @@ import (
 	volumetesting "k8s.io/kubernetes/pkg/volume/testing"
 	"k8s.io/kubernetes/pkg/volume/util/types"
 	"k8s.io/kubernetes/pkg/volume/util/volumehelper"
+	logpolicy "k8s.io/kubernetes/pkg/kubelet/log/policy"
 )
 
 func TestFindAndAddNewPods_FindAndRemoveDeletedPods(t *testing.T) {
@@ -540,6 +541,7 @@ func createDswpWithVolume(t *testing.T, pv *v1.PersistentVolume, pvc *v1.Persist
 			processedPods: make(map[types.UniquePodName]bool)},
 		kubeContainerRuntime:     fakeRuntime,
 		keepTerminatedPodVolumes: false,
+		logStatusProvider: logpolicy.NewPolicyStatusManager(),
 	}
 	return dswp, fakePodManager, fakesDSW
 }

